Students/Jacoby/Mechanum/subsystem/DriveTrain.java

```diff
@@ -0,0 +1,139 @@
+package Jacoby.Mechanum.subsystems;
+
+import java.time.format.ResolverStyle;
+
+import com.kauailabs.navx.frc.AHRS;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.Compressor;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.motorcontrol.MotorController;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+//import edu.wpi.first.wpilibj.motorcontrol.MotorController;
+//import edu.wpi.first.wpilibj.motorcontrol.Victor;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import Jacoby.Mechanum.Constants;
+
+public class DriveTrain extends SubsystemBase 
+{
+    
+
+    private static CANSparkMax frontLeftSparkMax = new CANSparkMax(Constants.MOTOR_FRONT_LEFT, MotorType.kBrushless);
+  private static CANSparkMax frontRightSparkMax = new CANSparkMax(Constants.MOTOR_FRONT_RIGHT, MotorType.kBrushless);
+  private static CANSparkMax backLeftSparkMax = new CANSparkMax(Constants.MOTOR_BACK_LEFT, MotorType.kBrushless);
+  private static CANSparkMax backRightSparkMax = new CANSparkMax(Constants.MOTOR_BACK_RIGHT, MotorType.kBrushless);
+
+
+  private static double[] motorCoefficients = {Constants.frontLeftMotorCoeff, Constants.frontRightMotorCoeff, Constants.backLeftMotorCoeff, Constants.backRightMotorCoeff};
+  
+  private static MecanumWrapperClass chassisDrive = new MecanumWrapperClass(frontLeftSparkMax, backLeftSparkMax, frontRightSparkMax, backRightSparkMax);
+
+  private static XboxController driverController = new XboxController(Constants.DRIVER_XBOX_CONTROLLER_PORT);
+
+  private static AHRS gyro = new AHRS();
+  private static double xInput;
+
+  private static double yInput;
+  private static double deltaZ;
+
+  private static double rInput;
+  private static boolean isMecanum = false;
+
+  private static DoubleSolenoid butterFlySolenoid = null;
+  private static Compressor phCompressor = null;
```

Comment:
These arnt being used and should be removed.

Reply:
Resolved